### PR TITLE
Change string contains operators to regex

### LINKED ISSUE #4817: Change `=~` and `!~` to match regex instead of substring
### Related problem

First: nushell is great! Keep up the great work!!

I have two related ideas:

1. It seems to me that the `where` and `find` commands are redundant: both of them have to do with filtering lists of things. It seems that `find` is intended for single-dimensional lists (i.e. `[moe larry curly]`) whereas `where` is intended for tables. Also, it seems that `find` can do regex, whereas `where` cannot.

    I'm sure there are technical reasons for having two commands, but as a user this is unintuitive. I feel that there should be a single command that can be used for filtering lists of things down to smaller lists of things.

2. The `where` command has the `=~` operator for finding substrings. This works great, but I think it could be better: the `=~` operator could be the "regex matching" operator. If this were the case, no functionality would be lost (substring matching is a subset of the functionality of regex) and much would be gained.

    Also, this is how `bash` works. I don't want `nu` to be `bash`, but there is a precedent that would make it easier for many people to learn and adopt nushell.

### Describe the solution you'd like

I propose abolishing the `find` command and merging its functionality into `where`. Funny enough, this is kind of supported at the moment:
```
[moe larry curly] | find l
```
seems to be equivalent to
```
[moe larry curly] | where $it =~ l
```
Of course, `where` does not have the regex functionality that `find` does. If `=~` became the regex operator as I described above, much of the functionality would be replaced. Then instead of
```
[moe larry curly] | find --regex 'l.*y'
```
you could do
```
[moe larry curly] | where $it =~ 'l.*y'
```
There is other functionality that `find` has:

`--insensitive`: easily replaced by using `(?i)` in your regex.
`--multiline`: easily replaced by piping the input into `lines` before filtering.
`--dotall`: seems "anti-nu" and best left to an external command like `grep` if you ever need to do this.
`--invert`: could be done using the `!~` operator in `where`.

As for the `--predicate` option I'm not so sure what to do with it. This is the realm of `filter` functions in other languages. But it seems redundant to have a `where` command and a `filter` command. But then again, there is a `reduce` command. `filter` may be the best way to go with this.

What do you think?

### Describe alternatives you've considered

_No response_

### Additional context and details

_No response_

### PR DESCRIPTION
**Pull request by rgwood**
# Description

This PR changes the `=~` and `!~` string-contains operators to regular expression operators. Closes #4817.

Both operators use [the regex crate's `is_match()` function](https://docs.rs/regex/latest/regex/struct.Regex.html#method.is_match) to determine whether a match for the pattern `rhs` exists in the string `lhs`. Specifically:

- `lhs =~ rhs` returns **true** if the string `lhs` contains a match for the regex pattern `rhs`, and **false** otherwise. 
- `lhs !~ rhs` returns **false** if the string `lhs` contains a match for the regex pattern `rhs`, and **true** otherwise. 

### Known issues

This change is mostly backwards-compatible with the previous `=~` and `!~` functionality, but it may break scripts that use `=~`/`!~` with right-hand expressions that are also valid regex patterns.

There is [a piece of dataframe code that has special handling for the `=~` operator](https://github.com/nushell/nushell/blob/2cb815b7b4cdc52b3934da459917d13ce89938a2/crates/nu-command/src/dataframe/values/nu_dataframe/between_values.rs#L369). I don't know the dataframe functionality well, so I've left it as is for now; until we sort this out `=~` will have a slightly different meanings in dataframe-land and regular Nu.

# Tests

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --all --all-features -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo build; cargo test --all --all-features` to check that all the tests pass

### CODE REVIEW

**Comment by elferherrera** [comment]
You can replace that operator in dfrs. That contains is used when you want to check if a string is contained in a columns with strings and it uses regex as well

**Comment by rgwood** [comment]
@elferherrera Just to confirm, what exactly do you mean by “replace that operator in dfrs”? 

Would you like me to change the code or comments in this PR?

**Comment by elferherrera** [comment]
Yeap. Otherwise the dataframe feature is not going to compile. I see that operator no longer has the contains type

File: crates/nu-command/src/dataframe/values/nu_dataframe/between_values.rs [hunk 1]
```diff
@@ -366,7 +366,8 @@ pub(super) fn compute_series_single_value(
                 rhs_span: right.span()?,
             }),
         },
-        Operator::Contains => match &right {
+        // TODO: update this to do a regex match instead of a simple contains?
+        Operator::RegexMatch => match &right {
```
Comment (rgwood) [inline crates/nu-command/src/dataframe/values/nu_dataframe/between_values.rs:370]: @elferherrera I've updated this already; is there another place I forgot? It builds successfully locally with `cargo build --features=extra`

Comment (elferherrera) [reply crates/nu-command/src/dataframe/values/nu_dataframe/between_values.rs:370]: That's all. It is the only place where that is used

**Comment by jntrnr** [comment]
nice!

**Comment by rgwood** [comment]
I'll update the website docs before next week's release.

**Pull request merged**